Vesuvius dnsmasq netboot

mehbark opened the pull request:
there's some minor problems

mehbark committed:
vesuvius: netboot prospit

mikhail729 committed:
corrected netboot configuration to avoid interface leakage and corrected addresses for dhcp clients on cluster vlan

mikhail729 committed:
corrected dhcp interface

mehbark committed:
hosts/vesuvius: change DNS domain from hephaestus to bastille

this is hugely important and a major blocker

kurisufriend commented on hosts/common/k3s-ports.nix:
```diff
@@ -0,0 +1,9 @@
+{
```
this should be encapsulated in common/k3s.nix

kurisufriend commented on hosts/common/k3s.nix:
```diff
@@ -0,0 +1,12 @@
+
+  services.k3s = {
+    enable = true;
+    role = "agent";
```
a sustainable way to make this singular role work for both worker nodes and the control plane could be to take in a variable assigned at import time. see: most of the stuff i wrote for zerocool.

kurisufriend commented on hosts/common/k3s.nix:
```diff
@@ -0,0 +1,12 @@
+    enable = true;
+    role = "agent";
+    token = "garbage secret";
+    serverAddr = "https://10.98.1.147:6443";
```
this should be a name, when we get that

mehbark committed:
hosts/common: refactor to remove k3s-ports.nix

mehbark committed:
hosts/prospit: set k3s role

mehbark committed:
hosts/vesuvius: become the k3s cluster init

mehbark committed:
hosts/common/k3s: refactor to take role and clusterInit args

mehbark committed:
hosts/prospit: be adopted by bastille

mehbark committed:
hosts/bastille: fix root password warning

mehbark committed:
hosts/bastille: automatically set hostname based on mac address

Ralakus commented on hosts/common/k3s.nix:
```diff
@@ -0,0 +1,17 @@
+
+    enable = true;
+    token = "garbage secret";
+    serverAddr = "https://10.98.1.147:6443";
```
Should be 10.98.3.2

mikhail729 reviewed (changes requested):
Incorrect cluster master IP address. Ideally the k3s server should only be listening on the cluster Vlan rather than all interfaces as well.

mehbark committed:
hosts/bastille: add some actual names to auto-hostname

there's still some possible issues with the script (hence the paranoid `echo` after `hostname`)

mehbark commented:
summary for myself tomorrow (yes i mean wednesday): there's some lingering bugs and misconfigurations + we need like fifty new `readme`s

kurisufriend committed:
fix(common/k3s): use cluster routing IP for vesuvius

kurisufriend committed:
Merge branch 'main' into vesuvius-dnsmasq-netboot

kurisufriend committed:
Merge branch 'main' into vesuvius-dnsmasq-netboot

kurisufriend committed:
!!breaking!! vesuvius/zfs: bind mount dataset /forge/nix to /nix, keeping the store off the root ssd

note that this is live on the machine, which *no longer has a native /nix*. this means that deploying any config which does not include the bind mount (e.g. master) will **break everything**. be duly warned.

mikhail729 committed:
hosts/vesuvius: opened port 53 for dns

kurisufriend committed:
flake: bump

kurisufriend committed:
feat(hosts/vesuvius): working freeIPA client

kurisufriend committed:
refactor(hosts/vesuvius): use roles/common stuff

kurisufriend committed:
feat(common/nfs): mount dirtycow stuff

kurisufriend committed:
feat(hosts/vesuvius): include dirtycow mounts

mikhail729 committed:
hosts/vesuvius: added dhcp-host entries for netboot nodes

mikhail729 committed:
hosts/bastille: disabled eno1 on lenovo nodes and removed auto-hostname script in favor of dhcp hostnames

kurisufriend committed:
fix(vesuvius/ipa): let "admins" group use passwordless sudo

kurisufriend committed:
docs(vesuvius/ipa): rm old comment about keytab

mikhail729 committed:
hosts/vesuvius: added nfs share and port for kubernetes storage

mikhail729 committed:
hosts/vesuvius: added static dnsmasq domain for vesuvius

mikhail729 committed:
hosts/{vesuvius,bastille}: opened ports for kubernetes intercluster communications

mikhail729 committed:
hosts/{bastille,common,vesuvius}: attempted k3s internal network ip address fix

mikhail729 committed:
hosts/vesuvius: added static ipv4 route from enp1s0f1 to shellshock network

mikhail729 committed:
hosts/common: fixed k3s command line parameters

mikhail729 committed:
hosts/{vesuvius, common}: added public addresses and nftables firewall

mikhail729 committed:
hosts/vesuvius: added caddy container as the kubernetes cluster reverse proxy and ipv6 firewall rules

mikhail729 committed:
kube-manifests/ephemeral: removed ephemeral.vtluug.org as host since it is behind a reverse proxy

mikhail729 committed:
hosts/vesuvius: changed caddy to use wildcard domains to the cluster

kurisufriend committed:
feat(vesuvius/caddy): dns acme validation w/ gandi api key

kurisufriend committed:
fix(vesuvius/caddy): move gandi secret

kurisufriend committed:
fix(vesuvius/caddy): add gandi plugin

mikhail729 committed:
hosts/vesuvius: format caddy.nix

kurisufriend committed:
fix(vesuvius/caddy): gandi plugin hash

kurisufriend committed:
Merge branch 'vesuvius-dnsmasq-netboot' of github.com:vtluug/construct into vesuvius-dnsmasq-netboot

mikhail729 committed:
hosts/vesuvius: fixed caddy proxy to bastille

mikhail729 committed:
flake.nix: removed depreciated arguments in favor of their replacements

mikhail729 committed:
update kubernetes manifests with correct configuration for the cluster

mikhail729 committed:
hosts/{vesuvius,bastille,common}: fixed nfs to cluster and fixed slow dns resolution

mikhail729 committed:
kube-manifests: split files into kubernetes configuration and kubernetes services

mikhail729 committed:
hosts/vesuvius: added caddy bindmount for persistent caddy state

mikhail729 committed:
hosts/vesuvius: changed caddy to use production lets encrypt and renamed search-staging to search

mikhail729 committed:
kube-manifests: pinned search to vesuvius

mikhail729 committed:
hosts/vesuvius: changed caddy to foward host header as is to cluster via https

mikhail729 committed:
kube-manifests: changed ingress hostnames to be direct subdomains of vtluug.org to avoid traefik rewriting redirect urls

mikhail729 committed:
kube-manifests: uncommented hack to re-enable https for opensearch

mikhail729 committed:
kube-manifests: fixed nfs and added bash

kurisufriend committed:
kube-manifests: add vtluug-site manifest

kurisufriend committed:
kube-manifests/vtluug-site: switch ingress to vtluug.org

kurisufriend committed:
vesuvius/caddy: port main site

kurisufriend committed:
keys: add my non-yubi pubkey

kurisufriend committed:
feat(k8s): sops secret framework, with mediawiki secrets as demo

kurisufriend committed:
feat(k8s): add wiki

kurisufriend committed:
fix(k8s/secrets): decrypted secrets should be 0600

kurisufriend committed:
feat(k8s/secrets): add bash-secrets

kurisufriend committed:
fix(k8s/wiki): muddle with permissions

kurisufriend committed:
fix(k8s/wiki): don't chown db (nfs squashroot: you can't!)

kurisufriend committed:
fix(vesuvius/caddy): switch handle_path /w/ to handle. caddy is annoying.

kurisufriend committed:
fix(k8s/wiki): image path rewrite

kurisufriend committed:
fix(vesuvius/caddy): gobblerpedia should use HTTP acme challenges

kurisufriend committed:
k8s/secrets: recipients file

kurisufriend committed:
fix(vesuvius/nfs): may god have mercy on our souls

kurisufriend committed:
fix(kube/wiki): pin maria to 12.2.2

kurisufriend committed:
fix(vesuvius/network): new ip

kurisufriend committed:
fix(vesuvius/caddy): use vesuvius for dns
